scripts/release/docker-build-all

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o xtrace
+
+cp -r /tmp/nomad-git /opt/gopath/src/github.com/hashicorp/nomad/.git
+
+cd /opt/gopath/src/github.com/hashicorp/nomad
+git checkout .
+
+make -f ./scripts/release/Makefile.linux \
+  "NOMAD_VERSION=${NOMAD_VERSION}" \
+  "PRERELEASE_TARGET=${PRERELEASE_TARGET}" \
+  "RELEASE_TARGET=${RELEASE_TARGET}" \
+  update_version build_releases
+
+cp -r /opt/gopath/src/github.com/hashicorp/nomad \
+    /tmp/artifacts/repo
```

Comment:
Why do we copy the entire repo instead of just the built binaries? Perhaps explain in a comment.

Reply:
I updated the doc.  We pass the whole repository, because it would contain some commits with generated files that we want to keep.